yt/data_objects/static_output.py

```diff
@@ -242,10 +243,10 @@ def __init__(
         self.no_cgs_equiv_length = False
 
         if unit_system == "code":
-            # create a fake MKS unit system which we will override later to
+            # create a fake unit system which we will override later to
             # avoid chicken/egg issue of the unit registry needing a unit system
             # but code units need a unit registry to define the code units on
-            used_unit_system = "mks"
+            used_unit_system = "mks" if self._use_mks_em_units else "cgs"
```

Comment:
So in theory it would be nice to use `default_units` for this instead, but there is the issue that the stream frontend would still have to change `default_units` if we used it to check for MKS EM units. I'm supposing that wasn't the intent--that it should not be edited on a per-object basis?

Reply:
@neutrinoceros any thoughts here?

Reply:
Sorry I missed that.

`default_units` is indeed intended as a constant class attribute, but its job is just to provide sane defaults, instances are free to use `units_override` to replace them partially or totally.
Couldn't we migrate this specialised behaviour in the stream frontend ?

Reply:
@neutrinoceros,

So `default_units` is for `units_override` only, correct? `units_override` was originally designed to provide units for datasets that weren't self-describing and didn't have any unit information in them--Athena datasets are notorious for this, for example. Because yt doesn't really work without assuming _some_ units, `units_override` provided sane defaults, so that you could at least do something with the dataset. We enabled `units_override` for all frontends, but it was always considered a solution of last resort, and if your dataset was self-describing you shouldn't even need it unless you were doing something weird.

In this use case for this PR, we need something that comes even before unit detection from the dataset, to let yt know it should expect something with dimensions of SI current in it when creating the code unit system. This has been a very tricky thing to manage  in yt while still allowing the end-user to switch to a different unit system, because there is a chicken-and-egg problem where we have to set up a code unit system before we can detect units, but only when we detect units can we know whether we need an SI current unit, which is info you need to create the code unit system in the first place. It's quite awful.

`default_units` seems like something quite different.

Reply:
> So default_units is for units_override only, correct?

yes that was the original intention.

Ok I think I understand this problem better now. It's unfortunate that complexity is going up but I reckon it may be the only way right now. Thanks a lot for going into the details, I appreciate it !

Reply:
NP--I'm going to copy the explanation I gave into the main text so the context is there.